src/ASM/ASMs2Dmx.C

```diff
@@ -175,8 +175,19 @@ bool ASMs2Dmx::generateFEMTopology ()
 {
   if (!surf) return false;
 
-  if (m_basis.empty())
+  if (m_basis.empty()) {
     m_basis = ASMmxBase::establishBases(surf, ASMmxBase::Type);
+
+    // we need to project on something that is not one of our bases
+    if (ASMmxBase::Type == ASMmxBase::REDUCED_CONT_RAISE_BASIS1 ||
+        ASMmxBase::Type == ASMmxBase::DIV_COMPATIBLE) {
+      auto vec2 = ASMmxBase::establishBases(surf,
+                                            ASMmxBase::FULL_CONT_RAISE_BASIS1);
+      projBasis = vec2.front();
+    }
```

Comment:
Do we need the `vec2`temporary here, or can we just do:
 `projBasis = ASMmxBase::establishBases(surf,ASMmxBase::FULL_CONT_RAISE_BASIS1).front();`

Reply:
done.

Reply:
And now the braces are not needed either, since only single-statement.

Reply:
removed.